### ISSUE TITLE
Pump IC collects non-source blocks

### ISSUE THREAD

**Issue by LadyCailin** [LadyCailin/TestIssueUploader#250]
CRAFTBOOK-2238 - Reported by kaitoyuuki on 2012-11-08 18:23:34 UTC

The pump IC allows users to obtain fluid blocks easily, and can also collect from non-source blocks. If a source block is connected to the flowing liquid, it will collect the source but give a flowing liquid block. This might be able to be exploited to duplicate lava.
I think it should be changed so that it uses buckets in chests rather than just collecting liquid blocks into them. Same goes for the spigot.

**Comment by LadyCailin** [comment]
Comment by me4502 on 2012-11-08 21:16:46 UTC

I don't know what you mean here, it checks if a block is a source block before picking it up... Also it underwent extensive testing to make sure it couldn't. Are you able to give a specific scenario in which this would occur?

**Comment by LadyCailin** [comment]
Comment by kaitoyuuki on 2012-11-08 22:26:16 UTC

I placed a pump sign on a block above flowing water. After I activated it, the chest above it had flowing water in it, whereas when I put it over a source block it has still water. I tested it, too, and have not been able to duplicate the liquids. At this point, I think it would still be a good idea to add an option in the config to require buckets.
